coriolisclient/tests/cli/test_endpoints.py

```diff
@@ -0,0 +1,594 @@
+# Copyright 2024 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import argparse
+from cliff import command
+from cliff import lister
+from cliff import show
+import ddt
+from unittest import mock
+
+from coriolisclient.cli import endpoints
+from coriolisclient import exceptions
+from coriolisclient.tests import test_base
+
+
+@ddt.ddt
+class EndpointTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Client Endpoints."""
+
+    @mock.patch.object(argparse, 'FileType')
+    def test_add_connection_info_args_to_parser(self, mock_FileType):
+        mock_parser = mock.Mock()
+        mock_add_argument = mock.Mock()
+        mock_parser.add_mutually_exclusive_group.return_value.add_argument = \
+            mock_add_argument
+
+        result = endpoints.add_connection_info_args_to_parser(mock_parser)
+
+        self.assertEqual(
+            mock_parser,
+            result
+        )
+        mock_parser.add_mutually_exclusive_group.assert_called_once()
+        mock_add_argument.assert_has_calls([
+            mock.call('--connection', help='JSON encoded connection data'),
+            mock.call('--connection-file', type=mock_FileType.return_value,
+                      help='Relative/full path to a file containing'
+                      ' the connection info in JSON format'),
+            mock.call('--connection-secret',
+                      help='The url of the Barbican secret '
+                      'containing the JSON connection info')
+        ])
+
+    @ddt.data(
+        {"conn": (None, None, None), "expected_result": None,
+         "raise_if_none": False},
+        {"conn": (None, None, None), "expected_result": "exception"},
+        {
+            "conn": ('{"conn_info": "mock_conn_info"}', None, None),
+            "expected_result": {"conn_info": "mock_conn_info"}
+        },
+        {
+            "conn": (None, '{"conn_info": "mock_conn_info"}', None),
+            "expected_result": {"conn_info": "mock_conn_info"}
+        },
+        {
+            "conn": (None, 'invalid', None),
+            "expected_result": "exception"
+        },
+        {
+            "conn": (None, None, "mock_secret"),
+            "expected_result": {"secret_ref": "mock_secret"}
+        },
+    )
+    def test_get_connnection_info_from_args(self, data):
+        mock_args = mock.MagicMock()
+        (mock_args.connection,
+            mock_args.connection_file.__enter__.return_value.read.return_value,
+            mock_args.connection_secret) = data["conn"]
+        if (data["conn"][1] is None):
```

Comment:
Please remove paranthesis:
```suggestion
        if data["conn"][1] is None:
```